pypdf/_writer.py

```diff
@@ -3052,7 +3052,14 @@ def _insert_filtered_annotations(
                         outlist.append(ano.clone(self).indirect_reference)
                 else:
                     d = cast("ArrayObject", d)
-                    p = self._get_cloned_page(d[0], pages, reader)
+                    try:
+                        p = self._get_cloned_page(d[0], pages, reader)
+                    except TypeError:
+                        # There are cases where the IndirectObject is not None, but d[0] will fail:
```

Comment:
You basically want to check for `is_null_or_none(d[0].get_object())`? Having this explicitly is preferred over more costly exception handling.